homework-1/cash_register.py

```diff
@@ -0,0 +1,66 @@
+class CashRegister:
+
+    def __init__(self):
+        self.total_items = {}  # {'item': 'price'}
+        self.total_price = 0
+        self.discount = 0
+
+    def add_item(self, item_price: dict):
+        self.total_items.update(item_price)
+        for key, value in item_price.items():
+            self.total_price += value
+            print(f"The {key} with the price of £{value} has been added to the cart.")
+
+    def remove_item(self, item):
+        self.total_items.pop(item)
+        print(f"{item.capitalize()} has been removed from the cart.")
+
+    def apply_discount(self, discount_percentage: float):
+        self.discount = round((self.total_price * discount_percentage), 2)
+        print(f"The amount of £{self.discount} will be discounted from the total price.")
+
+    def get_total(self):
+        self.total_price -= self.discount
+        print(f"The total price to be paid is £{self.total_price}.")
```

Comment:
I like this in principle, but notice that the discount is applied multiple times if this method is run repeatedly, which isn't quite right. It would also have been good to return the final total. Perhaps calculating it using `self.total_price` and `self.discount` without modifying `self.total_price` directly might have been better